app.py

```diff
@@ -18,6 +18,160 @@
 """This is the main script of the template project."""
 
 from template.version import __version__
+import logging
+import sys
+from datetime import date
+from datetime import datetime
+from typing import Dict
+from typing import Any
+from typing import Optional
+from urllib.parse import urlparse
 
-if __name__ == "__main__":
-    print(f"A template project with Thoth integration, v{__version__}.")
+import requests
+import click
+import yaml
+from thoth.common import init_logging
+from thoth.storages import SolverResultsStore
+
+init_logging()
+_LOGGER = logging.getLogger("thoth.prescriptions.gh_release_notes")
+_LOGGER.setLevel(logging.INFO)
+_DATE_FORMAT = "%Y-%m-%d"
+
+__component_version__ = f"{__version__}"
+
+
+def get_source_repos(*, start_date: Optional[date], end_date: Optional[date]) -> Dict[str, Any]:
+    """Get source URLs of github repos."""
+    store = SolverResultsStore()
+    store.connect()
+
+    git_source_repos = {}
+    for document_id, doc in store.iterate_results(start_date=start_date, end_date=end_date, include_end_date=True):
+        if not doc["result"]["tree"]:
+            continue
+        _LOGGER.debug("Processing solver document %r", document_id)
+
+        metadata = doc["result"]["tree"][0]["importlib_metadata"]["metadata"]
+        name = metadata.get("Name")
+        if not name:
+            continue
+        url_candidates = [metadata.get("Home-page")]
+        for url in metadata.get("Project-URL") or []:
+            url_candidates.append(url.rsplit(",", maxsplit=1)[-1].strip())
+
+        for url in url_candidates:
+            if not url or not url.startswith("https://github.com"):
+                continue
+            _LOGGER.debug("Processing URL: %r", url)
+            url_path_parts = urlparse(url).path.split("/")[1:]
```

Comment:
If parsing fails (e.g. malformed URL in the dataset), the job will fail. Might worth to report the exception/error and continue with the rest.

Reply:
Is this something we want to write as a metric so that we can have an alert/github issue created from it or is it an exception we just want to have in sentry?

Reply:
> Is this something we want to write as a metric so that we can have an alert/github issue created from it or is it an exception we just want to have in sentry?

I think having a sentry report for this is good enough for this case. The report should state what URL failed to parse so we can eventually fix/adjust the parsing logic here.